transcribe.py

```diff
@@ -0,0 +1,44 @@
+import json
+
+import typer
+import torch
+
+from transformers import pipeline
+from typing_extensions import Annotated
+from typing import Optional
+
+# TODO: Check if optimum is installed
+# TODO: Add a function to check hardware/ GPU and choose an appropriate model
+
+
+def main(
+    file_name: str,
+    device_id: str,
+    transcript_path: str,    
+    model_name: str,
+    task: str,
+    language: str,
+):
+    pipe = pipeline(
+        task="automatic-speech-recognition",
+        model=model_name,
+        torch_dtype=torch.float16,
+        device=f"cuda:{device_id}",
+    )
+
+    pipe.model = pipe.model.to_bettertransformer()
+
+    outputs = pipe(
+        file_name,
+        chunk_length_s=30,
+        batch_size=24,
```

Comment:
My GPU was using 12 GB of RAM while transcribing. It has more RAM, I assume these parameters are a good balance between memory and performance?

Reply:
Yeah! Basically beyond this point we reach diminishing returns. This is the most I could get out of a Colab T4 and I assume an average joe has that.

Reply:
Yes, they can tweak it manually if they want to experiment :)